diag_manager/fms_diag_field_object.F90

```diff
@@ -0,0 +1,965 @@
+     logical, allocatable, private                    :: registered        !< true when registered
+     logical, allocatable, private                    :: mask_variant      !< If there is a mask variant
+     logical, allocatable, private                    :: do_not_log        !< .true. if no need to log the diag_field
+     logical, allocatable, private                    :: local             !< If the output is local
+     TYPE(time_type), private                         :: init_time         !< The initial time
+     integer,          allocatable, private           :: vartype           !< the type of varaible
+     character(len=:), allocatable, private           :: varname           !< the name of the variable
+     character(len=:), allocatable, private           :: longname          !< longname of the variable
+     character(len=:), allocatable, private           :: standname         !< standard name of the variable
+     character(len=:), allocatable, private           :: units             !< the units
+     character(len=:), allocatable, private           :: modname           !< the module
+     character(len=:), allocatable, private           :: realm             !< String to set as the value
+                                                                           !! to the modeling_realm attribute
+     character(len=:), allocatable, private           :: interp_method     !< The interp method to be used
+                                                            !! when regridding the field in post-processing.
+                                                            !! Valid options are "conserve_order1",
+                                                            !! "conserve_order2", and "none".
+     integer, allocatable, dimension(:), private      :: frequency         !< specifies the frequency
+     integer, allocatable, dimension(:), private      :: output_units
+     integer, allocatable, private                    :: t
+     integer, allocatable, private                    :: tile_count        !< The number of tiles
+     integer, allocatable, dimension(:), private      :: axis_ids          !< variable axis IDs
+     class(diagDomain_t), pointer,   private          :: domain            !< Domain
+     INTEGER                         , private        :: type_of_domain    !< The type of domain ("NO_DOMAIN",
+                                                                           !! "TWO_D_DOMAIN", or "UG_DOMAIN")
+     integer, allocatable, private                    :: area, volume      !< The Area and Volume
+     class(*), allocatable, private                   :: missing_value     !< The missing fill value
+     class(*), allocatable, private                   :: data_RANGE(:)     !< The range of the variable data
+    contains
+!     procedure :: send_data => fms_send_data  !!TODO
+! Get ID functions
+     procedure :: get_id => fms_diag_get_id
```

Comment:
why are there two procedures for getting the id?

Reply:
Who knows, lol